modules/osint/discord_search.py

```diff
@@ -0,0 +1,82 @@
+"""
+OWASP Maryam!
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import re
+
+meta = {
+	'name': 'Discord Search',
+	'author': 'Divya Goswami',
+	'version': '1.0',
+	'description': 'Search the name of any user or server on discord [from disboard and discordhub]',
+	'sources': ('discord',),
+	'options': (
+		('user', None, False, 'User flag (search for a user)', '-u', 'store', str),
+		('server', None, False, 'Server flag (search for a server)', '-s', 'store', str),
+		('limit', 3, False, 'Search limit(number of pages, default=3)', '-l', 'store', int),
+		('engine', 'discord', False, 'Engine used is discord', '-e', 'store', str),
+		('thread', 2, False, 'The number of engine that run per round(default=2)', '-t', 'store', int),
+	),
+	'examples': ('discord_search -u <QUERY> --output', 'discord_search -s <QUERY> -l 5')
+	}
+
+PAGES = ''
+
+def search(self, name, q, q_formats, limit, count):
+	global PAGES, LINKS
+	engine = getattr(self, name)
+	name = engine.__init__.__name__
+	varnames = engine.__init__.__code__.co_varnames
+	if 'limit' in varnames:
+		attr = engine(q, limit)
+	attr.run_crawl()
+	PAGES += attr.pages
+
+def module_api(self):
+	user = self.options['user']
+	server = self.options['server']
+	if user and server:
+		self.error("Choose to search only one among user and server.")
+		return
+	elif user:
+		if user.isdigit():
+			query = f'{user}_u'
+		else:
+			self.error("Enter user's 4 digit tag only [ #<Number> ]")
```

Comment:
as well. You can only show the errors in module_run. Or returning something like this:
{'error': 'msg'}

Reply:
```
┬─[divya at racharch in ~/a/g/OWASP-Maryam]─[G:discord]
╰──> λ ./maryam -e discord_search -u wolf
[!] Enter user's 4 digit tag only [ #<Number> ].
[*] None
```